File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.py
class Solution:
    def countGoodSubstrings(self, s: str) -> int:
        '''
            Time complexity = O(n), 
            Space complexity = O(1)
        '''

        left, right = 0, 2
        n = len(s)
        count = 0

        while(right < n) :
            if s[left] != s[left + 1] and \
            s[left + 1] != s[left + 2] and \
            s[left + 2] != s[left] :

                count += 1
            
            left += 1
            right += 1
        
        return count

        

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
class Solution {
    public int countGoodSubstrings(String s) {
        /*
            Two pointer approach :-
            Time complexity = O(n), 
            Space complexity = O(1)
        */
        
        int n = s.length();
        int left = 0, right = 2;
        int countGoodSubStr = 0;

        while(right < n){
            char sCh = s.charAt(left);
            char sChNext = s.charAt(left + 1);
            char sChNextNext = s.charAt(left + 2);

            if(sCh != sChNext && sChNext != sChNextNext && sChNextNext != sCh)
            {
                countGoodSubStr++;
            }

            left++;
            right++;
        }

        return countGoodSubStr;
    }
}